src/lighteval/tasks/registry.py

```diff
@@ -87,74 +84,143 @@ def __init__(self, cache_dir: Optional[str] = None, custom_tasks: Optional[Union
+
+        return configs
 
     @property
     @lru_cache
-    def task_registry(self):
+    def task_registry(self) -> dict[str, LightevalTaskConfig]:
         """
         Returns:
             dict[str, LazyLightevalTask]: A dictionary mapping task names (suite|task) to their corresponding LightevalTask classes.
 
         Example:
             {
-                "lighteval|arc_easy": lambda: LightevalTask(name="lighteval|arc_easy", ...)
+                "lighteval|arc_easy": LightevalTaskConfig(name="arc_easy", suite="lighteval", ...),
             }
         """
-
-        # Import custom tasks provided by the user
         custom_tasks_registry = {}
         custom_tasks_module = []
-        TASKS_TABLE = []
+        custom_task_configs = []
+
         if self._custom_tasks is not None:
-            custom_tasks_module.append(create_custom_tasks_module(custom_tasks=self._custom_tasks))
+            custom_tasks_module.append(Registry.create_custom_tasks_module(custom_tasks=self._custom_tasks))
         if can_load_extended_tasks():
             for extended_task_module in AVAILABLE_EXTENDED_TASKS_MODULES:
                 custom_tasks_module.append(extended_task_module)
         else:
             logger.warning(CANNOT_USE_EXTENDED_TASKS_MSG)
 
         for module in custom_tasks_module:
-            TASKS_TABLE.extend(module.TASKS_TABLE)
-            # We don't log the tasks themselves as it makes the logs unreadable
+            custom_task_configs.extend(module.TASKS_TABLE)
             logger.info(f"Found {len(module.TASKS_TABLE)} custom tasks in {module.__file__}")
 
-        if len(TASKS_TABLE) > 0:
-            custom_tasks_registry = create_lazy_tasks(meta_table=TASKS_TABLE, cache_dir=self._cache_dir)
+        if len(custom_task_configs) > 0:
+            custom_tasks_registry = Registry.create_task_config_dict(meta_table=custom_task_configs)
+
+        default_tasks_registry = Registry.create_task_config_dict()
 
-        default_tasks_registry = create_lazy_tasks(cache_dir=self._cache_dir)
         # Check the overlap between default_tasks_registry and custom_tasks_registry
         intersection = set(default_tasks_registry.keys()).intersection(set(custom_tasks_registry.keys()))
         if len(intersection) > 0:
             logger.warning(
                 f"Following tasks ({intersection}) exists both in the default and custom tasks. Will use the custom ones on conflict."
             )
 
-        # Custom tasks overwrite defaults tasks
         return {**default_tasks_registry, **custom_tasks_registry}
 
+    def taskinfo_selector(self, tasks: str) -> dict[str, list[dict]]:
+        """
+        Converts a input string of tasks name to task information usable by lighteval.
+
+        Args:
+            tasks (str): A string containing a comma-separated list of tasks definitions in the
+                format "task_definition|few_shot|truncate_few_shots" or a path to a file
```

Comment:
(current comment gives the impression we define nested few_shot calls)
```suggestion
                format: "task_definition", where it can be